### PR TITLE
Disable Noble balance polling when connecting Kepler wallet

### PR DESCRIPTION
**Pull request by Ryz0nd**
Hello dYdX Team!

This PR addresses the ongoing issue of Noble USDC transfer tx signing requests persisting when connecting the Keplr wallet on the dYdX web platform.

We are planning to implement a separate USDC deposit modal, and we want the signing request to occur only when this modal is activated.

So, we have added the `cosmosWalletConnected` state to prevent the `pollNobleBalance` method from executing when the Kepler wallet is connected. 

If there are better approaches, we would appreciate your feedback.

Thank you.

https://github.com/dydxprotocol/v4-abacus/assets/53689198/5f1cf209-8508-4547-8e2d-fe821def8748

### CODE REVIEW

File: src/commonMain/kotlin/exchange.dydx.abacus/state/manager/AsyncAbacusStateManager.kt [hunk 1]
```diff
@@ -144,6 +145,14 @@ class AsyncAbacusStateManager(
             }
         }
 
+    override var cosmosWalletConnected: Boolean? = false
+        set(value) {
+            field = value
+            ioImplementations.threading?.async(ThreadingType.main) {
```
Comment (prashanDYDX) [inline src/commonMain/kotlin/exchange.dydx.abacus/state/manager/AsyncAbacusStateManager.kt:151]: this should be `ThreadingType.abacus`

File: src/commonMain/kotlin/exchange.dydx.abacus/state/manager/StateManagerAdaptor.kt [hunk 2]
```diff
@@ -195,6 +195,13 @@ open class StateManagerAdaptor(
             }
         }
 
+    var cosmosWalletConnected: Boolean? = false
+        set(value) {
+            if (field != value) {
+                field = value
+            }
+        }
```
Comment (prashanDYDX) [inline src/commonMain/kotlin/exchange.dydx.abacus/state/manager/StateManagerAdaptor.kt:203]: custom setter not necessary here

File: src/commonMain/kotlin/exchange.dydx.abacus/state/v2/manager/StateManagerAdaptorV2.kt [hunk 3]
```diff
@@ -259,6 +259,14 @@ internal class StateManagerAdaptorV2(
             accounts.accountAddress = value
         }
 
+    internal var cosmosWalletConnected: Boolean?
+        get() {
+            return return accounts.cosmosWalletConnected
```
Comment (prashanDYDX) [inline src/commonMain/kotlin/exchange.dydx.abacus/state/v2/manager/StateManagerAdaptorV2.kt:264]: extra return here

File: src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/AccountSupervisor.kt [hunk 4]
```diff
@@ -113,6 +113,13 @@ internal open class AccountSupervisor(
             }
         }
 
+    var cosmosWalletConnected: Boolean? = false
+        internal set(value) {
+            if (field != value) {
+                field = value
+            }
+        }
```
Comment (prashanDYDX) [inline src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/AccountSupervisor.kt:121]: custom setter not necessary here

**Review by prashanDYDX — APPROVED** [review]

**Comment by Ryz0nd** [comment]
@prashanDYDX Thanks! Is there anything else that needs to be done for this PR to be merged?

**Pull request closed without merging**